doc/hacking.rst

```diff
@@ -223,4 +223,24 @@ To adhere to this loading scheme, we must generate our rules as part
 of the callback that creates targets in that directory. See the ``Gen_rules``
 module for how this callback is constructed.
 
+Bootstrapping
```

Comment:
I think I would put this near the top since it is an important gotcha. Perhaps also some emphasis on running `./dune.exe` when building targets in the repo?

Reply:
done